contracts/wallet/BaseWallet.sol

```diff
@@ -117,42 +119,43 @@ contract BaseWallet {
-    function invoke(address _target, uint _value, bytes calldata _data) external moduleOnly returns (bytes memory _result) {
+    function invoke(address _target, uint _value, bytes calldata _data) external virtual moduleOnly returns (bytes memory _result) {
         bool success;
         // solium-disable-next-line security/no-call-value
-        (success, _result) = _target.call.value(_value)(_data);
+        (success, _result) = _target.call{value: _value}(_data);
         if (!success) {
             // solium-disable-next-line security/no-inline-assembly
             assembly {
-                returndatacopy(0, 0, returndatasize)
-                revert(0, returndatasize)
+                returndatacopy(0, 0, returndatasize())
+                revert(0, returndatasize())
             }
         }
         emit Invoked(msg.sender, _target, _value, _data);
     }
 
     /**
-     * @dev This method makes it possible for the wallet to comply to interfaces expecting the wallet to
-     * implement specific static methods. It delegates the static call to a target contract if the data corresponds
-     * to an enabled method, or logs the call otherwise.
+     * @dev This method delegates the static call to a target contract if the data corresponds
+     * to an enabled module, or logs the call otherwise.
      */
-    function() external payable {
-        if (msg.data.length > 0) {
-            address module = enabled[msg.sig];
-            if (module == address(0)) {
-                emit Received(msg.value, msg.sender, msg.data);
-            } else {
-                require(authorised[module], "BW: must be an authorised module for static call");
-                // solium-disable-next-line security/no-inline-assembly
-                assembly {
-                    calldatacopy(0, 0, calldatasize())
-                    let result := staticcall(gas, module, 0, calldatasize(), 0, 0)
-                    returndatacopy(0, 0, returndatasize())
-                    switch result
-                    case 0 {revert(0, returndatasize())}
-                    default {return (0, returndatasize())}
-                }
+    fallback() external payable {
+        address module = enabled[msg.sig];
+        if (module == address(0)) {
+            emit Received(msg.value, msg.sender, msg.data);
+        } else {
+            require(authorised[module], "BW: must be an authorised module for static call");
+            // solium-disable-next-line security/no-inline-assembly
+            assembly {
+                calldatacopy(0, 0, calldatasize())
+                let result := staticcall(gas(), module, 0, calldatasize(), 0, 0)
+                returndatacopy(0, 0, returndatasize())
+                switch result
+                case 0 {revert(0, returndatasize())}
+                default {return (0, returndatasize())}
             }
         }
     }
+
+    receive() external payable {
+        emit Received(msg.value, msg.sender, msg.data);
```

Comment:
I think we can remove this event emission and leave `receive()` empty. If I'm not mistaken `receive()` here will only ever be called in our tests.

Reply:
I've made `Receive` only emit on non-zero values https://github.com/argentlabs/argent-contracts/pull/114/commits/12c79ec27dba5802f5665672cc11d0786b186f28